libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -150,6 +150,16 @@ AudioSegmentSampleView WaveClip::GetSampleView(
       start + TimeToSamples(mTrimLeft), length, mayThrow);
 }
 
+AudioSegmentSampleView WaveClip::GetSampleView(
```

Comment:
I see that before this commit, `AudioSegmentSampleView` was constructed in only one place outside tests, in `Sequence`, and always passing `size_t length` to the `sampleCount` argument.

If `AudioSegmentSampleView` really eliminated use of type `sampleCount` and only used `size_t`, it would move the need to use `as_size_t` to the point of construction.

There are new places where it is constructed, all of them in `WaveTrack` and added by this commit.

Reply:
I agree, I should be consistent with myself. A sample view is typically used to store a temporary buffer, and the total amount should never exceed what `size_t` can hold. Shall I simplify this?

Reply:
would be better